src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -129,6 +130,37 @@ public Icon(Stream stream, int width, int height) : this()
             Initialize(width, height);
         }
 
+        private Icon(SerializationInfo info, StreamingContext context)
+        {
+            _iconData = (byte[])info.GetValue("IconData", typeof(byte[])); // Do not rename (binary serialization)
+            _iconSize = (Size)info.GetValue("IconSize", typeof(Size)); // Do not rename (binary serialization)
+
+            if (_iconSize.IsEmpty)
+            {
+                Initialize(0, 0);
```

Comment:
Doesn't IsEmpty just mean _iconSize.Width == 0 && _iconSize.Height == 0?  I'm wondering why this if branch is needed at all.